avro_schema/frontend.lua

```diff
@@ -304,67 +309,80 @@ copy_schema = function(schema, context, ns, open_rec)
                 end
                 if type(xfields) ~= 'table' then
-                    copy_schema_error('Record "fields" must be a list')
+                    copy_schema_error(context,
+                        'Record "fields" must be a list')
                 end
                 if #xfields == 0 then
-                    copy_schema_error('Record type must have at least one field')
+                    copy_schema_error(context,
+                        'Record type must have at least one field')
                 end
                 res.fields = {}
                 local fieldmap = {}
                 for fieldno, xfield in ipairs(xfields) do
-                    ptr = fieldno
+                    local ptr = fieldno
                     local field = {}
                     preserve_user_fields(xfield, field, context)
                     res.fields[fieldno] = field
                     if type(xfield) ~= 'table' then
-                        copy_schema_error('Record field must be a list')
+                        copy_schema_error(context,
+                            'Record field must be a list')
                     end
                     local xname = xfield.name
                     if not xname then
-                        copy_schema_error('Record field must have a "name"')
+                        copy_schema_error(context,
+                            'Record field must have a "name"')
                     end
                     xname = tostring(xname)
                     if not validname(xname) then
-                        copy_schema_error('Bad record field name: %s', xname)
+                        copy_schema_error(context,
+                            'Bad record field name: %s', xname)
                     end
                     if fieldmap[xname] then
-                        copy_schema_error('Record contains field %s twice', xname)
+                        copy_schema_error(context,
+                            'Record contains field %s twice', xname)
                     end
                     fieldmap[xname] = fieldno
                     field.name = xname
                     local xtype = xfield.type
                     if not xtype then
-                        copy_schema_error('Record field must have a "type"')
+                        copy_schema_error(context,
+                            'Record field must have a "type"')
                     end
                     field.type = copy_schema(xtype, context, ns, open_rec)
                     if open_rec[type_tag(field.type)] then
-                        local path, n = {}
-                        for i = 1, 1000000 do
-                            local _, res = debug.getlocal(i, LOCAL_VAR_POS_res)
-                            if res == field.type then
-                                n = i
-                                break
-                            end
+                        local path = {}
```

Comment:
Why the code from `copy_schema_location_info` does not do the trick here? Both are needed to get a path.

Reply:
That change is not straightforward, they differ in several ways.
Can be done in an extra patch.

Reply:
If there is still reason to leave it separate, then so. I don't insist.